src/solvers/flattening/boolbv_bswap.cpp

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************\
+
+Module:
+
+Author: Michael Tautschnig
+
+\*******************************************************************/
+
+#include "boolbv.h"
+
+#include <util/config.h>
+#include <util/invariant.h>
+
+bvt boolbvt::convert_bswap(const exprt &expr)
+{
+  if(expr.id() != ID_bswap || expr.operands().size() != 1)
```

Comment:
I don't like that we can have expressions with their own id and no associated class (inheriting from `exprt`). In particular it is not documented anywhere how these bswap expressions are structured.